deb/ch3/spark-intro/notebooks/01_intro.ipynb

```diff
@@ -0,0 +1,261 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# What is Spark?\n",
+    "To quote the Spark website:\n",
+    "\"Apache Spark™ is a unified analytics engine for large-scale data processing.\" But what does this mean? We can think of Spark as a powerful egine that is able to perform a wide arrary of data manipulation and analytical tasks while also being able to scale to truly big data sets. It's the number \\#1 distributed data processing engine in the world and it's offered as a service on all Cloud vendors as part of their \"Big Data Analytical\" package. Spark can be used for batch transformation, such as transforming large number of data files or images; OR is used for stream processing like live website or app logs.\n",
+    "\n",
+    "So how are we going to use this engine? While Spark itself is written is Scala, Pyspark provides us Python programmers a library to run Spark. In order to install the Pyspark module, run the following in your Python environment:"
+   ]
+  },
+  {
+   "cell_type": "raw",
+   "metadata": {},
+   "source": [
+    "pip install pyspark"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## SparkContext: how you interact with your data\n",
+    "Now that we have Pyspark rinstalled, let's restart the kernel for this notebook and figure out how to actually use it. Our first goal is to be able to load in our data. Becaues Spark is designed to work with data in HDFS and other big data storage tools as well as local data, it has its own set of data interface objects. As A SparkContext is the most basic of these for connecting to your data. In larger scale operations, a SparkContext could be configured to work across clusters or other cloud computing setups but for now, we're just using the little local cluster that is the machine that is running this notebook. To do this, we create a SparkContext object with the `'local[*]'` options. The `[*]` allows this SparkContext to have access to all local cores; you can manully set this number lower if you would like to limit the context. "
```

Comment:
```suggestion
    "Now that we have PySpark installed, let's restart the kernel for this notebook and figure out how to actually use it. Our first goal is to be able to load in our data. Because Spark is designed to work with data in HDFS and other big data storage tools as well as local data, it has its own set of data interface objects. A SparkContext is the most basic of these for connecting to your data. In larger scale operations, a SparkContext could be configured to work across clusters or other cloud computing setups but for now, we're just using the little local cluster that is the machine that is running this notebook. To do this, we create a SparkContext object with the `'local[*]'` options. The `[*]` allows this SparkContext to have access to all local cores; you can manually set this number lower if you would like to limit the context. "
```